### PR TITLE
RD-2529 Add API-GW v2 support

### PR DESCRIPTION
**Pull request by doriaviram**

### CODE REVIEW

**Comment by efimk-lu** [comment]
@doriaviram please ask Saar to review it as well

File: src/lumigo_tracer/parsers/utils.py [hunk 1]
```diff
@@ -196,10 +196,19 @@ def _is_supported_http_method(event: dict):
         and "headers" in event  # noqa
         and "requestContext" in event  # noqa
         and event.get("requestContext", {}).get("elb") is None  # noqa
+        or event.get("version", "") == "2.0"  # noqa
```
Comment (saartochner) [inline src/lumigo_tracer/parsers/utils.py:199]: can you add parenthesis?
It's confusing...

File: src/lumigo_tracer/parsers/utils.py [hunk 2]
```diff
@@ -196,10 +196,19 @@ def _is_supported_http_method(event: dict):
...
+        and "headers" in event  # noqa
     )  # noqa
 
 
 def parse_http_method(event: dict):
+    version = event.get("version")
+    if version == "2.0":
```
Comment (saartochner) [inline src/lumigo_tracer/parsers/utils.py:206]: don't you prefer `startwith("2.")` so we will not miss minors?

File: src/lumigo_tracer/parsers/utils.py [hunk 3]
```diff
@@ -213,6 +222,18 @@ def parse_http_method(event: dict):
     return result
 
 
+def _parse_http_method_v2(event: dict):
+    result = {
+        "triggeredBy": "apigw",
+        "httpMethod": event.get("requestContext", {}).get("http", {}).get("method"),
+        "resource": event.get("requestContext", {}).get("http", {}).get("path"),
+        "messageId": event.get("requestContext", {}).get("requestId", ""),
```
Comment (saartochner) [inline src/lumigo_tracer/parsers/utils.py:230]: why the default here is empty string and not None like the rest? 😂

File: src/test/unit/parsers/test_parser.py [hunk 4]
```diff
@@ -18,3 +18,29 @@ def test_get_parser_check_headers():
     headers = http.client.HTTPMessage()
     headers.add_header("x-amzn-requestid", "1234")
     assert get_parser(url, headers) == ServerlessAWSParser
+
+
+def test_get_parser_apigw():
+    url = "execute-api.us-west-2.amazonaws.com"
```
Comment (saartochner) [inline src/test/unit/parsers/test_parser.py:24]: can you put here a real url?

File: src/test/unit/parsers/test_utils.py [hunk 5]
```diff
@@ -261,6 +261,30 @@ def test_recursive_json_join(d1, d2, result):
...
+                    "http": {
+                        "method": "GET",
+                        "path": "/default/nodejs-apig-function-1G3XMPLZXVXYI",
+                    },
+                    "requestId": "JKJaXmPLvHcESHA=",
+                    "stage": "default",
+                },
+            },
+            {
+                "triggeredBy": "apigw",
```
Comment (saartochner) [inline src/test/unit/parsers/test_utils.py:280]: It looks like we're missing information here - 
maybe we want to change to `apigw2`?

File: src/lumigo_tracer/parsers/utils.py (hunk 3 above)
Comment (doriaviram) [reply src/lumigo_tracer/parsers/utils.py:230]: This is how its implemented in V1, I don't know if it's for a reason

File: src/test/unit/parsers/test_parser.py (hunk 4 above)
Comment (doriaviram) [reply src/test/unit/parsers/test_parser.py:24]: 👍

File: src/test/unit/parsers/test_utils.py (hunk 5 above)
Comment (doriaviram) [reply src/test/unit/parsers/test_utils.py:280]: For the backend, it's regular API-gw.
Just other places to look for our interesting fields

File: src/lumigo_tracer/parsers/utils.py (hunk 1 above)
Comment (doriaviram) [reply src/lumigo_tracer/parsers/utils.py:199]: 👍

**Pull request merged**